src/cc2olx/olx.py

```diff
@@ -1,65 +1,145 @@
-                    }
-                if type == "link":
-                    type, details = convert_link_to_video(details)
-                if type == "link":
-                    type = "html"
-                    details = {
-                        "html": "<a href='{}'>{}</a>".format(details["href"], details.get("text", "")),
-                    }
-                if type == "html":
-                    child = self.doc.createElement("html")
-                    txt = self.doc.createCDATASection(details["html"])
-                    child.appendChild(txt)
-                elif type == "video":
-                    child = self.doc.createElement("video")
-                    child.setAttribute("youtube", "1.00:" + details["youtube"])
-                    child.setAttribute("youtube_id_1_0", details["youtube"])
-                elif type == 'lti':
-                    child = self._create_lti_node(details)
-                else:
-                    raise Exception("WUT")
+                content_type, details = self._get_content(element_data)
+                children = self._create_olx_nodes(content_type, details)
             else:
-                child = self.doc.createElement(tags[0])
-            if "title" in dd:
-                child.setAttribute("display_name", dd["title"])
-            elt.appendChild(child)
-            if "children" in dd:
-                self._add_olx_nodes(child, dd["children"], tags[1:])
+                children = [self.doc.createElement(tags[0])]
+
+            for child in children:
+                if "title" in element_data:
+                    child.setAttribute("display_name", element_data["title"])
+
+                element.appendChild(child)
+
+                if "children" in element_data:
+                    self._add_olx_nodes(child, element_data["children"], tags[1:])
+
+    def _get_content(self, element_data):
+        """
+        Gets content type and details from element's data.
+        """
+
+        content_type = None
+        details = None
+
+        if "identifierref" in element_data:
+            idref = element_data["identifierref"]
+            content_type, details = self.cartridge.get_resource_content(idref)
+
+        if content_type is None:
+            content_type = self.HTML
+            details = {
+                "html": "<p>MISSING CONTENT</p>",
+            }
+
+        if content_type == self.LINK:
+            content_type, details = process_link(details)
+
+        return content_type, details
+
+    def _create_olx_nodes(self, content_type, details):
+        """
+        Based on content type and element details creates appropriate
+        child nodes.
+        """
+
+        nodes = []
+
+        if content_type == self.HTML:
+            child = self.doc.createElement("html")
+            txt = self.doc.createCDATASection(details["html"])
+            child.appendChild(txt)
+
+            nodes.append(child)
+
+        elif content_type == self.VIDEO:
```

Comment:
If I follow correctly, if we have a video on some other site besides youtube, the conversion in `process_link()` doesn't need to happen, and then we will just treat the video as an HTML link, and it will be handled in the block just above?

Reply:
Yes, @azarembok, that's correct. I think about it like about previews in popular messengers. They usually can render content from some popular platforms like youtube, but everything else is displayed like regular link.

I also found that course from edX  partners widely uses iframes in places when it's needed to show video from another platform, and we render these iframes correctly.

Reply:
👍